### ISSUE TITLE
Allow categories without IsCongruentForMorphisms

### ISSUE THREAD

**Issue by mohamed-barakat** [homalg-project/CAP_project#391]
I am building categories without the ability to decide the mathematical equality of morphisms. For this, it suffices to disable the following two final derivations for `IsCongruentForMorphisms` and `IsEqualForMorphisms` which default to `ReturnFail`:

https://github.com/homalg-project/CAP_project/blob/55817a0d356ec41ced14089f3874d7e00045b776/CAP/gap/DerivedMethods.gi#L3176

I have disabled them in my branches and things are still working correctly for all the CAP-based packages I tested. Is there something I am missing that necessitates such final derivations?

**Comment by mohamed-barakat** [comment]
If I do not disable both of them then more methods get derived (e.g., `IsZeroObject`, `IsIsomorphism`, etc.) which upon invocation return the error

```
Error, Method for a property did not return true or false
not in any function at *stdin*:1
you can 'return true;' or 'return false;'
```
since as properties they return the output of `IsCongruentForMorphisms` which is `fail`.

**Comment by mohamed-barakat** [comment]
To be more precise, I just need to disable `IsCongruentForMorphisms` as it is the one used for deriving other methods. However, this final derivation

https://github.com/homalg-project/CAP_project/blob/55817a0d356ec41ced14089f3874d7e00045b776/CAP/gap/DerivedMethods.gi#L3192

will reinstall `IsCongruentForMorphisms` using `IsEqualForMorphisms`.

**Comment by sebastianpos** [comment]
Just to clarify: there always has to be an `IsEqualForMorphisms`, i.e., the technical equality of morphisms. Technically, it is needed for type checking. Mathematically, it is needed for the specifications: equal input yields equal output. If you don't know your notion of `IsEqualForMorphisms`, there is simply no way to implement a category whose underlying basic operations are mathematical functions.

Concerning `IsCongruentForMorphisms`, yes, we could simply not trigger any final derivation for `IsCongruentForMorphisms` in order to allow categories without mathematical equality. But then we lose the comfort that if the notions of `IsCongruentForMorphisms` and `IsEqualForMorphisms` coincide (like in our matrix category), you only have to provide one of them. So if you delete the final derivations, you have to go through all packages and possibly rectify the installations of `IsCongruentForMorphisms` and `IsEqualForMorphisms`.

**Comment by mohamed-barakat** [comment]
Yes, right. I'll change the title of the issue an close it with a PR in the future.

**Comment by zickgraf** [comment]
Fixed by 1c6b65d2227dc9cba780cafbd415bf770bfff012.
